### DISCUSSION TITLE
VSCode - Powershell module errors.

### DISCUSSION THREAD

**Discussion by win-scripter** [microsoft/vscode-discussions#944] [Extension Development QnA]
I am facing issues with VSCode running Powershell modules.

VSCode details below
Version: 1.84.1 (system setup)
Commit: 2b35e1e6d88f1ce073683991d1eff5284a32690f
Date: 2023-11-06T12:35:59.022Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19044

The modules work fine with PowershellISE but when I run it through VSCode I get the error below.

PS C:\Users\admin> connect-azAccount
connect-azAccount : Method 'get_SerializationSettings' in type 'Microsoft.Azure.Management.Internal.Resources.ResourceManagementClient' from assembly 
'Microsoft.Azure.PowerShell.Clients.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
At line:1 char:1
+ connect-azAccount
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Connect-AzAccount], TypeLoadException
    + FullyQualifiedErrorId : System.TypeLoadException,Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand

I have tried uninstalling VSCode, updating the Azure modules, etc. without much help.

Thanks in advance

**Comment by theReynald** [comment]
Hi @win-scripter.

Sounds like the issue you're experiencing with the `connect-azAccount` command from the Azure PowerShell module, seems to be related to a mismatch or conflict between module versions or dependencies.  

So this might take some trial an error but try the following steps:

1. **Check PowerShell Versions:**
   - Ensure that the version of PowerShell being used in VSCode is the same as the one in PowerShell ISE where the modules work fine. You can check the PowerShell version in VSCode by running `$PSVersionTable` in the terminal.

2. **Check Azure PowerShell Module Versions:**
   - Run `Get-Module -ListAvailable Az*` in both PowerShell ISE and VSCode to compare the versions of Azure PowerShell modules installed.

3. **Reinstall Azure PowerShell Module:**
   - Since the issue seems to be with the Azure PowerShell module, try reinstalling it. You can do this by running `Uninstall-Module -Name Az -AllVersions` followed by `Install-Module -Name Az`.

4. **Run VSCode as Administrator:**
   - Sometimes, running VSCode with elevated privileges can resolve module loading issues. Right-click on the VSCode shortcut and select "Run as administrator".

5. **Update VSCode and Extensions:**
   - Ensure that VSCode and all its extensions, especially the PowerShell extension, are up to date. You can check for updates in VSCode under the Extensions tab.

6. **Profile Script Conflict:**
   - Check if there is any code in your PowerShell profile script that might be causing conflicts. You can open your profile script in VSCode with `code $PROFILE`.

7. **Dependency Conflict:**
   - There might be a conflict with another module or a different version of a dependency. Investigate if other modules loaded in your VSCode session might conflict with the Azure module.

8. **Check Execution Policy:**
   - Ensure your execution policy allows running the scripts/modules. You can check this with `Get-ExecutionPolicy` and set it appropriately with `Set-ExecutionPolicy`.

9. **Use Integrated Console:**
   - Make sure you are using the PowerShell Integrated Console in VSCode, which is part of the PowerShell extension, as it is specifically designed to work with PowerShell in VSCode.

10. **Check for Known Issues:**
    - Sometimes, these errors are due to known issues with specific versions of modules or the PowerShell extension in VSCode. Check the GitHub repositories or forums for any reported issues and potential fixes.

Let me know if this helped at all.

**Comment by olguzzar** [comment]
Hi @win-scripter, this forum is specifically for extension developers, so I'm going to close this discussion as off-topic. Thanks!
